pages/platform/private-registry/configure-oidc-provider-authentication/guide.en-au.md

```diff
@@ -0,0 +1,288 @@
+It makes it easy to secure applications and services with little to no code.
+More information can be found here: [Official Keycloak documentation](https://www.keycloak.org/documentation.html).
+
+**This guide will explain how to configure the authentication of your OVHcloud Managed Private Registry service to use an OIDC provider.**
+
+## Requirements
+
+- An OVHcloud [Managed Private Registry](/pages/platform/private-registry/creating-a-private-registry).
+- The URL and login/password of your private registry.
+- an OIDC provider, like a KeyCloack instance and credentials to access it.
+
+Follow the official guide if you want to [install Keycloack on an OVHcloud Managed Kubernetes cluster](/pages/platform/kubernetes-k8s/installing-keycloak).
+
+## Instructions
+
+### Configure ythe OIDC provider through the API
+
+#### The API Explorer
+
+To simplify things, we are using the [API Explorer](https://api.ovh.com/) which allows to explore, learn and interact with the API in an interactive way.
+
+Log in to the API Explorer using your OVHcloud account credentials.
+
+![Log in to the API Explorer](images/kubernetes-quickstart-api-ovh-com-001.png){.thumbnail}
+
+If you go to the [Kubernetes section](https://api.ovh.com/console/#/cloud/project/%7bservicename%7d/kube~get) of the API Explorer, you will see the available endpoints:
+
+![Kubernetes section of the API Explorer](images/kubernetes-quickstart-api-ovh-com-002.png){.thumbnail}
+
+#### API endpoints
+
+- Configure an OIDC provider for your OVHcloud Managed Private Registry:
+
+Do a POST HTTP request and fill the input fields with the keycloack/OIDC provider information.
+
+> [!api]
+>
+> @api {POST} /cloud/project/{serviceName}/containerRegistry/{registryID}/openIdConnect
+>
+
+**Input:**
+
+```
+serviceName: <your service name>
+registryID: <the ID of the registry>
+OIDCPost:
+  deleteUsers: true
+  provider:
+    adminGroup: harbor-admin //it's my admin group in keycloack
+    clientSecret: <keycloack client secret>
+    endpoint: <keyclock endpoint> //ending with the realm, for example: https://xxx.yyy.com/realms/mpr 
+    groupsClaim: <group in keycloack>
+    name: keycloack //for example
+    scope: openid,profile,email,offline_access
+    userClaim:
+    verifyCert: true //if you want to check the keycloack SSL/TLS certificate
+```
+
+**Result:**
+
+```json
+null
+```
+
+Go to the [Harbor documentation](https://goharbor.io/docs/2.8.0/administration/configure-authentication/oidc-auth/) to see the details of the needed information.
+
+> [!primary]
+>
+> If you forget to enable the `deleteUsers` option, you will have an error: `{ "class": "Client::BadRequest", "message": "4 users exist, set parameter deleteUsers to delete them" }`. Indeed, when a new Private Registry is created, several users are created. So you need to enable the `deleteUsers` option in order to remove existing local users and enable OIDC authentication.
```

Comment:
New users are created by customer, if he didn't created new users through Control panel or API there will be no user.
I would have add a warning to warn that when the option deleteUsers is checked, it will delete all existing users.